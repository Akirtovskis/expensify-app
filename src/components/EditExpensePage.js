import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense} from '../actions/expenses';

//Refactor EditExpense to class based component
//Setup mapDispatchToProps editExpense removeExpense

//beforeEach()
//Should render edit expense page (SNAPSHOT)
//Should handle edit expense (SPIES)
//Should handle remove expense (REMOVE)
export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onRemove = (expense) => {
    this.props.startRemoveExpense({id:this.props.expense.id});
    this.props.history.push('/');
  }
  render(){
    return (
      <div>
        <ExpenseForm
          expense = {this.props.expense}
          onSubmit = {this.onSubmit}
        />
        <button onClick={this.onRemove}
        >Remove</button>
      </div>
    );
  };
}

const mapStateToProps = (state, props) => {
  return {
    expense : state.expenses.find((expense)=> expense.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense : (id, expense)=>dispatch(editExpense(id, expense)),
  startRemoveExpense : (data)=>dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
