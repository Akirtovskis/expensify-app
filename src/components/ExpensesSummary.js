import React from 'react';
import numeral from 'numeral';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';

export const ExpensesSummary = (props) => {
  const length = props.expenses.length;
  const total = expensesTotal(props.expenses);
  return (
    <div>
      Viewing : {(length===1 )? (length + " expense ") :(length + " expenses ") }
      totalling : {numeral(total/100).format('$0,0.00')}
    </div>
  );
}

const mapStateToProps = (state)=> {
  return {
    expenses : selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
