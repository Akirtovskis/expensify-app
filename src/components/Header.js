import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <h3>
      <NavLink to='/' activeClassName='is-active' exact={true}>Home </NavLink>
      <NavLink to='/create' activeClassName ='is-active'>Create Expense </NavLink>
    
    </h3>
  </header>
);

export default Header;
