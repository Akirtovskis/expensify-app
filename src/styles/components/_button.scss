.button {
  background:$blue;
  border-radius:3px;
  color:white;
  display:inline-block;
  font-size:$font-size-large;
  font-weight:300;
  line-height:1;
  padding:$s-size;
  text-decoration:none;
}

.button--logout {
  background:none;
  border:none;
}

.button--secondary {
  background:#888888;
}
