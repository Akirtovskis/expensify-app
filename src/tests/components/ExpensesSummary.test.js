import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly with 1 expense', () => {
  const wrapper= shallow(<ExpensesSummary expenses={[expenses[0]]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpenseSummary correctly with multiple expenses', () => {
  const wrapper= shallow(<ExpensesSummary expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});
