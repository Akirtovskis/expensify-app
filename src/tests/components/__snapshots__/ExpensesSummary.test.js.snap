// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  Viewing : 
  3 expenses 
  totalling : 
  $1,141.95
</div>
`;

exports[`Should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  Viewing : 
  1 expense 
  totalling : 
  $1.95
</div>
`;

exports[`Should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  Viewing : 
  0 expenses 
  totalling : 
  $0.00
</div>
`;
